test/test_reductions.py

```diff
@@ -461,6 +461,82 @@ def test_mode(self, device):
             # input unchanged
             self.assertEqual(x, x0, atol=0, rtol=0)
 
+    def _test_mode_intervals(self, shape, intervals, device, v = 1):
+        x = torch.arange(0, shape[0] * shape[1], device = device)
+        x[v] = x.numel()
+        x = x.resize_(shape)
+
+        # Set the value of each interval to the mode "v"
+        for (beg, end) in intervals:
+            x[:, beg:end] = v
+
+        values, indices = torch.mode(x, -1, False)
+
+        # Check whether the indices that correspond to each mode found are within
+        # one of the defined intervals
```

Comment:
Just checking here that
```
(x.gather(1, indices.unsqueeze(1)).t() == values).all() 
```
is enough and more readable.

Reply:
That makes sense.